crates/common/consensus/src/deneb/beacon_state.rs

```diff
@@ -81,11 +82,37 @@ use crate::{
     withdrawal::Withdrawal,
 };
 
+fn serialize_u8_as_quoted_list<S>(
+    value: &VariableList<u8, U1099511627776>,
+    serializer: S,
+) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let string_vec: Vec<String> = value.iter().map(|v| v.to_string()).collect();
+    string_vec.serialize(serializer)
+}
+fn quoted_u8_var_list<'de, D>(deserializer: D) -> Result<VariableList<u8, U1099511627776>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let string_vec: Vec<String> = Vec::deserialize(deserializer)?;
+    let bytes_result: Result<Vec<u8>, _> = string_vec
+        .into_iter()
+        .map(|s| s.parse::<u8>().map_err(serde::de::Error::custom))
+        .collect();
+
+    let bytes = bytes_result?;
+    Ok(VariableList::from(bytes))
```

Comment:
```suggestion
fn quoted_u8_var_list<'de, D>(deserializer: D) -> Result<VariableList<u8, U1099511627776>, D::Error>
where
    D: Deserializer<'de>,
{
    let string_vec = Vec::<String>::deserialize(deserializer)?;
    let bytes = string_vec
        .into_iter()
        .map(|s| s.parse::<u8>().map_err(serde::de::Error::custom))
        .collect::<Result<Vec<_>, _>>()?;

    Ok(VariableList::new(bytes))
```

`VariableList::new` we should use new and handle the error